packages/react-aria-components/src/Menu.tsx

```diff
@@ -200,6 +206,79 @@ function MenuInner<T extends object>({props, collection, menuRef: ref}: MenuInne
     }
   }, [leftOffset, popoverContainer]);
 
+  let {id: menuId} = menuProps;
+  useEffect(() => {
+    if (register) {
+      register((e: ReactKeyboardEvent) => {
```

Comment:
The register and clearVirtualFocus that happen here feel like they should be in the menu hooks, would we be comfortable added `register` and `inputValue` (or something along those lines) as something the hooks accept?